### DISCUSSION TITLE
Pagination of files in a commit

### DISCUSSION THREAD

**Discussion by frink182** [hub4j/github-api#1667] [Q&A]
Is there a way to retrieve more than the default GH limit of 300 files per commit? I'm using `GHCommit.getFiles()` which only pulls back 300 files at most.

**Comment by bitwiseman** [comment]
https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit

> Note: If there are more than 300 files in the commit diff, the response will include pagination link headers for the remaining files, up to a limit of 3000 files. Each page contains the static commit information, and the only changes are to the file listing.

Except `hub4j` doesn't support pagination at this time: 
https://github.com/hub4j/github-api/blob/9801b86b8aff903aaee4b7b0983a616ee646db0d/src/main/java/org/kohsuke/github/GHCommit.java#L418-L421

I've just filed #1669 .  Do you have time to work on this?

**Reply by frink182** [reply]
Thanks for confirming the defect. I'll see how I get on with fixing it.

**Reply by bitwiseman** [reply]
@frink182  It's not the easiest change. I've added a note to the issue with implementation suggestion.

**Reply by frink182** [reply]
Thanks, that info does help.

**Reply by frink182** [reply]
@bitwiseman I have a fix ready, so I'm going to need access to hub4j-test-org please.

**Reply by bitwiseman** [reply]
@frink182 I've sent you an invite.
